packages/components/src/custom-select-control-v2/types.ts

```diff
@@ -40,14 +51,100 @@ export type CustomSelectProps = {
 	/**
 	 * The size of the control.
 	 *
-	 * @default 'default'
+	 * @default default
 	 */
-	size?: 'default' | 'small';
+	size?: 'compact' | 'default' | 'small';
```

Comment:
```suggestion
	size?: 'compact' | 'default';
```

Like @ciampo suggested, we can probably get away with just these two sizes for the V2.